docs/spec/auth/vesting.md

```diff
@@ -0,0 +1,88 @@
+## Vesting
+
+### Intro and Requirements
+
+This paper specifies changes to the auth and bank modules to implement vesting accounts for the Cosmos Hub. 
+The requirements for this vesting account is that it should be capable of being initialized during genesis with
+a starting balance X coins and a vesting blocktime T. The owner of this account should be able to delegate to validators 
+and vote with locked coins, however they cannot send locked coins to other accounts until those coins have been unlocked. 
+The vesting account should also be able to spend any coins it receives from other users or from fees/inflation rewards. 
+Thus, the bank module's `MsgSend` handler should error if a vesting account is trying to send an amount that exceeds their 
+unlocked coin amount.
+
+### Implementation
+
+##### Vesting Account implementation
+
+```go
+type VestingAccount interface {
+    Account
+    AssertIsVestingAccount() // existence implies that account is vesting.
+}
+
+// Implements Vesting Account
+// Continuously vests by unlocking coins linearly with respect to time
+type ContinuousVestingAccount struct {
+    BaseAccount
+    OriginalCoins sdk.Coins
+    ReceivedCoins sdk.Coins
+    StartTime     int64
+    EndTime       int64
+}
+
+func (vacc ContinuousVestingAccount) ConvertAccount() BaseAccount {
+    if T > vacc.EndTime {
+        // Convert to BaseAccount
+    }
+}
```

Comment:
Take a look at the other specs - I don't think we need to use explicit golang code here for functions as is done (although I like the use of golang for types) - other places we haven't used golang for functions

Reply:
I think some pseudocode is great though?

Reply:
@jaekwon yeah pseudocode is good! I just mean we don't need golang code. Better to convert golang code to pseudocode for funcs just to stay consistent with the rest of the specs